akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala

```diff
@@ -705,9 +706,9 @@ abstract class GraphStageLogic private[stream] (val inCount: Int, val outCount:
         setHandler(in, new Reading(in, n - pos, getHandler(in))((elem: T) => {
           result(pos) = elem
           pos += 1
-          if (pos == n) andThen(result)
-        }, () => onClose(result.take(pos))))
-      } else andThen(result)
+          if (pos == n) andThen(result.toSeq)
+        }, () => onClose(result.take(pos).toSeq)))
+      } else andThen(result.toSeq)
```

Comment:
Not trivial to see that this is harmless from this diff, but if you verified then :+1:

Reply:
Without this explicit conversion there was an implicit conversion `Predef.genericWrapArray` which calls `WrappedArray.make(array)`  which then constructs a `WrappedArray` so there is actually no added alloc overhead, the overhead was already there (checked in 2.12).